From 8a8fd824582d35bd2ffee3fdc6fbbbed45974bd1 Mon Sep 17 00:00:00 2001
From: Haohui Mai <wheat9@apache.org>
Date: Fri, 12 Dec 2014 11:51:17 -0800
Subject: [PATCH 1778/2748] HDFS-7517. Remove redundant non-null checks in
 FSNamesystem#getBlockLocations. Contributed by
 Haohui Mai.

(cherry picked from commit d9f31c6472d6c7264e2d685a058dec3501a73ffd)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: Ic161d771f0be3623e92fd3df81c2ac01d761ec79
---
 .../hadoop/hdfs/server/namenode/FSNamesystem.java  |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
index 655ab8f..8c8c84a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java
@@ -1879,10 +1879,6 @@ LocatedBlocks getBlockLocations(String clientMachine, String src,
 
     logAuditEvent(true, "open", src);
 
-    if (res == null) {
-      return null;
-    }
-
     if (res.updateAccessTime()) {
       writeLock();
       final long now = now();
-- 
1.7.9.5

